### DISCUSSION TITLE
Support for Nedis Doorbell searching protocol (lsc doorbell)

### DISCUSSION THREAD

**Discussion by Hoeby** [arendst/Tasmota#13616] [Ideas]
I have a Nedis Doorbell, which sends 433mhz when pushing the button.
The Nedis is one example, there a multi manufactors which selling this doorbell.
The housing is different, but inside it is the same. Like the LSC doorbell from Action
Trying to get it working with the rc-switch library, but without succes.

To get some information, i used the simple-rc-scanner and got the next code
`432,59856,456,380,452,384,452,380,452,380,456,380,452,380,452,384,452,380,452,384,452,380,452,384,452,380,452,384,448,388,448,384,448,388,448,384,448,384,452,384,448,384,448,388,448,388,444,388,448,388,448,384,448,388,444,388,448,388,444,388,444,392,444,388,444,1224,444,392,860,388,1280,388,444,392,444,1640,444,392,856,396,1272,392,444,392,444,1224,440,1200,444,808,440,12904,440,1224,444,1224,1276,392,1276,392,444,1224,444,1228,436,1228,440,1228,440,1232,436,1228,1276,392,1276,392,440,1228,1272,396,1272,396,440,1228,440,1228,436,1232,1272,396,1272,396,436,1232,1272,396,436,1232,436,1232,436,1232,436,1232,1268,400,1268,400,1268,400,432,1236,432,1236,432,1236,432,1232,436,1232,1272,400,1268,396,1268,400,436,1232,436,1236,1240,400,432,1236,432,1236,1268,400,1264,404,432,1240,428,1232,436,1236,1264,400,436,1236,428,1236,1268,400,1268,404,428,1236,1264,404,432,1236,1264,404,432,1236,432,1236,1264,404,1264,404,1264,404,432,1236,428,1240,1264,404,428,1240,428,1240,1260,408,1264,404,428,1240,428,1240,1260,408,1260,408,428,1240,1260,408,1260,408,1260,408,424,1240,1264,404,1264,404,428,1240,428,1240,428,1240,428,1240,1260,408,424,1244,428,1240,428,1240,428,1212,428,1240,428,1240,428,1240,1260,408,428,1240,428,1240,428,1240,1260,408,428,1240,428,1240,428,1236,428,1240,428,1240,428,1240,1264,404,428,1240,428,1240,428,1240,1260,408,1260,408,432,1236,1260,408,428,1240,1260,408,428,1240,424,1244,428,1240,1260,408,1260,408,1260,408,1260,408,424,1240,428,1240,428,1240,428,1240,428,1240,428,1240,428,1240,428,1240,428,14004,
`

When i look in the code, it looks like there is a pre-sample. Which is a challenge for rc-switch
![second sample](https://user-images.githubusercontent.com/40724351/140808019-6c7f77cc-7631-490d-923b-08948b4b894a.JPG)

When i push multi times. I get protocol 12, 16, 17 as response, with every time different data.
![data](https://user-images.githubusercontent.com/40724351/140808572-4d1c6a4f-32b9-46e8-aa50-b0a7fac739ab.JPG)

With a competitor from Tasmota, that works with this type of code, it gives me 64 bits (without sync).
But i don't want to use the competitor, i want to use Tasmota.

Is it possible to get this 433mhz protocol to work with the rc-switch library?
Or is it a problem, because of the design of rc-switch?
Can somebody get me on the right road, how to get this working?

**Comment by sfromis** [comment]
rc-switch is not part of Tasmota, but a separate project. https://github.com/sui77/rc-switch

**Reply by Jason2866** [reply]
There is no competition. Every Open Source Solution is geat to have.
To mention others here is no problem. Maybe the way it is done there can be implemented in Tasmota.

**Reply by Hoeby** [reply]
@sfromis, i know rc-switch is a separate project. But not from sui77/rc-switch. Maybe the documentation says so, but it is not. The code is more like the one from https://github.com/1technophile/rc-switch
But there is no option for discussion on that github library.

@Jason2866, that was not what i mentioned, that there is competition. I didn't know if it is allright to mention others, used the wrong words to explain that, sorry. 

I have used esphome, that one works with this kind of code. They call it also rc-switch, but for me it looks completly different to the rc-switch which is in tasmota

**Reply by Jason2866** [reply]
Correct, we use a modified variant of the 1technophile fork. And yes, all open source projects can be mentioned :-) All work in Opne Source is done for everyone for making "things" better.
With a link to the code that works for you, you make everyone easier to have a look.

**Comment by Hoeby** [comment]
Esphome works with "remote_receiver" and "remote_transmitter" 
[receiver link part](https://github.com/esphome/esphome/tree/dev/esphome/components/remote_receiver)
[transmitter link part](https://github.com/esphome/esphome/tree/dev/esphome/components/remote_transmitter)
If i got it right, this is to buffer the receiver/transmitter data.
Then it is transferred to the protocol (IR/RF). [link](`https://github.com/esphome/esphome/tree/dev/esphome/components/remote_base`)

When i push the nedis doorbell. I get a raw dump of rc_switch.
Therefor i went to the "remote base" and then the rc_switch protocol.
On line 247, there is rc_switch dump.
https://github.com/esphome/esphome/blob/dev/esphome/components/remote_base/rc_switch_protocol.cpp#L247

I am not a programmer, that makes it not easier to understand al those single files, which make it 1 program.
But i have fun in it to try and get it to work. But with this one i need help to understand and where to start to get it integrated with tasmota

**Comment by Hoeby** [comment]
Why is this marked as answered?
I unmarked it 2 times, because rcswitch is not the used library. But 2 times it is marked again that the answer is: rcswitch is not part of tasmota.

Tasmota uses a modified fork. When i go to rcswitch with the question. Then they say, it is not the same as rcswitch and also mark it as answered.

Where to go with the question?

**Reply by Jason2866** [reply]
The question is placed correctly here. The rcswitch part is only modified for Tasmota needs from the 1technophile fork which is a fork of rcswitch. Tasmota dev crew are not experts in rf protocols.
It may take some time if someone have a look.
